tinygrad/tensor.py

```diff
@@ -1023,11 +1023,28 @@ def pad(self, arg:Tuple[Optional[Tuple[sint, sint]], ...], value:float=0.0) -> T
   #   3. Out of bounds Tensor indexing results in 0
   #     - e.g: Tensor([1, 2, 3])[Tensor([4, 3, 2])] -> [0, 0, 3] index 4 and 3 are out of bounds
   def _getitem(self, indices, v: Optional[Tensor] = None) -> Tensor:
+    def _fix_slice(x: slice) -> slice:
+      # convert slice scalar Tensor slice parameters to int
+      def process_component(component):
+        if isinstance(component, Tensor):
+          if component.numel() > 1 or not dtypes.is_int(component.dtype): 
+            raise TypeError("only integer tensors of single element can be converted to a slice index")
+          return component.item()
```

Comment:
it should not have a `.item` here, as it becomes data dependent

Reply:
Got it
But then the question is, should tinygrad support single-element tensors in slices, which will involve some more clever refactoring, or not and just throw an type err that this operation is not supported?